src/rulesets/Base/broadcast.jl

```diff
@@ -318,7 +322,7 @@ rrule(::typeof(broadcasted), ::typeof(complex), x::Number) = rrule(complex, x) |
 
 function unbroadcast(x::Base.AbstractArrayOrBroadcasted, dx_raw)
     dx = unthunk(dx_raw)
-    N = ndims(dx)
+    N = _ndims(dx)
```

Comment:
This one has a test, lazy broadcasting of `-` means this can be called with `dx::Tuple`